src/query/functions/binary/common.go

```diff
@@ -109,8 +109,7 @@ func combineMetaAndSeriesMeta(
 	tags := models.NewTags(leftTags.Len(), leftTags.Opts)
 
 	for _, t := range leftTags.Tags {
-		name := string(t.Name)
-		if otherTag, ok := otherTags[name]; ok {
+		if otherTag, ok := otherTags[string(t.Name)]; ok {
```

Comment:
should otherTags use a genny map too?

Reply:
Updated it to do so; would be a smaller map that would be used much less often than the others so was less concerned about putting a genny map here initially